src/app/components/HighlightedText/text-matching.ts

```diff
@@ -0,0 +1,246 @@
+
+// -----------------------------------------
+
+/**
+ * Determine whether a simple character is alphanumerical or not
+ */
+export const isCharAlphaNumerical = (char: string): boolean => {
+  const code = char.charCodeAt(0)
+  return (
+    (code > 47 && code < 58) || // numeric (0-9)
+    (code > 64 && code < 91) || // upper alpha (A-Z)
+    (code > 96 && code < 123)
+  ) // lower alpha (a-z)
+}
+
+/**
+ * Escape a character to be used in a regexp-based text search, also considering normalization
+ */
+export const escapeCharForPCRE = (char: string, options: NormalizerOptions) => {
+  const { diacriticSensitive = false, caseSensitive = false } = options
+  if (diacriticSensitive) {
+    if (isCharAlphaNumerical(char) || caseInsensitiveDiacriticNormalizationTable.has(char)) {
+      // TODO: test if we need to manually do case-insensitive in this case
+      return char
+    } else {
+      return `\\${char}`
+    }
+  } else {
+    const mapping = caseSensitive
+      ? caseSensitiveDiacriticRegexpTable.get(char)
+      : caseInsensitiveDiacriticRegexpTable.get(char)
+    return mapping || (isCharAlphaNumerical(char) ? char : `\\${char}`)
+  }
+}
+
+/**
+ * Escape a string so that it becomes a valid PCRE regexp, also considering normalization
+ *
+ * (I.e. escape all the non-alphanumerical characters.)
+ */
+export const escapeTextForPCRE = (input: string, options: NormalizerOptions = {}) =>
+  input
+    .split('')
+    .map(char => escapeCharForPCRE(char, options))
+    .join('')
+
+/**
+ * Prepare a Mongo Query expression for doing a text search, also considering normalization
+ */
+export const getMongoRegexpSearch = (pattern: string, options: NormalizerOptions = {}) => {
+  const { caseSensitive = false } = options
+  return {
+    $regex: escapeTextForPCRE(pattern, options),
+    $options: caseSensitive ? 'm' : 'im',
+  }
+}
+
+/**
+ * Normalize a character for text search
+ */
+export const normalizeCharForSearch = (char: string, options: NormalizerOptions = {}) => {
+  const { caseSensitive = false, diacriticSensitive = false } = options
+  const stage1 = diacriticSensitive
+    ? char
+    : (caseSensitive
+        ? caseSensitiveDiacriticNormalizationTable.get(char)
+        : caseInsensitiveDiacriticNormalizationTable.get(char)) || char
+  const stage2 = caseSensitive ? stage1 : stage1.toLowerCase()
+  return stage2
+}
+
+/**
+ * A basic text normalizer function
+ */
+export const normalizeTextForSearch = (text: string, options: NormalizerOptions = {}) =>
+  text
+    .split('')
+    .map(char => normalizeCharForSearch(char, options))
+    .join('')
+
+/**
+ * Identify pattern matches within a corpus, also considering normalization
+ */
+export const findTextMatch = (
+  rawCorpus: string | null | undefined,
+  search: (string | undefined)[],
+  options: NormalizerOptions = {},
+): MatchInfo => {
+  let matchStart: number
+  let normalizedPattern: string
+  const normalizedCorpus = normalizeTextForSearch(rawCorpus || '', options)
```

Comment:
I think Intl.Collator / localeCompare is meant to replace this

Reply:
Yes, there is probably a better solution. (I just grabbed the code from a several years old public repo of mine.)

The bigger question is (which we should focus on before more technical review) is whether we want this feature or now.

Reply:
> I think Intl.Collator / localeCompare is meant to replace this

As described above, as far as I can understand, currently' it doesn't. However, we could involve another library, or we could remove text normalization entirely. As far as I am concerned, the current implementation satisfies my requirements.

What do you think, @lukaw3d  ?

Reply:
For now, I moved all the text normalization code into a separate module.

Reply:
All custom code has been removed now.